Tests/FluentAssertions.Shared.Specs/ObjectAssertionSpecs.cs

```diff
@@ -454,6 +454,68 @@ public void When_object_is_assignable_to_the_expected_type_it_should_cast_the_re
+        [TestMethod]
+        public void When_asserting_an_object_is_assignable_to_its_base_type_using_type_instance_it_should_succeed()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            var someObject = new DummyImplementingClass();
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act / Assert
+            //-----------------------------------------------------------------------------------------------------------
+            someObject.Should().BeAssignableTo(typeof(DummyBaseClass));
+        }
+
+        [TestMethod]
+        public void When_asserting_an_object_is_assignable_to_an_implemented_interface_type_using_type_instance_it_should_succeed()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            var someObject = new DummyImplementingClass();
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act / Assert
+            //-----------------------------------------------------------------------------------------------------------
+            someObject.Should().BeAssignableTo(typeof(IDisposable));
+        }
+
+        [TestMethod]
+        public void When_asserting_an_object_is_assignable_to_an_unrelated_type_using_type_instance_it_should_fail_with_a_descriptive_message()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            var someObject = new DummyImplementingClass();
+            Type assignableCheckType = typeof(DateTime);
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act / Assert
+            //-----------------------------------------------------------------------------------------------------------
+            someObject.Invoking(
+                x => x.Should().BeAssignableTo(assignableCheckType, "because we want to test the failure {0}", "message"))
+                .ShouldThrow<AssertFailedException>()
+                .WithMessage(string.Format(
+                    "Expected object to be assignable to {1} because we want to test the failure message, but {0} is not",
+                    typeof(DummyImplementingClass), assignableCheckType));
```

Comment:
I prefer to write these as:

```
//-----------------------------------------------------------------------------------------------------------
// Act
//-----------------------------------------------------------------------------------------------------------
Action act = () => someObject.Should().BeAssignableTo(assignableCheckType, "because we want to test the failure {0}", "message");

//-----------------------------------------------------------------------------------------------------------
// Assert
//-----------------------------------------------------------------------------------------------------------
act.ShouldThrow<AssertFailedException>()
   .WithMessage($"*assignable to {assignableCheckType}*failure message*{typeof(DummyImplementingClass)} is not*"));
```

This solves two problems:
1. `Invoking` is an old construct and is IMO a little less readable.
2. The exact exception message is irrelevant, as long as the important parts are there.